File: ta_service/models/analysis.py
from __future__ import annotations

from pydantic import BaseModel


class AnalysisTaskStatusResponse(BaseModel):
    taskId: str
    status: str
    symbol: str | None = None
    teamId: str | None = None
    currentStep: str | None = None
    message: str | None = None
    elapsedTime: int | None = None
    remainingTime: int | None = None


class CreateAnalysisTaskRequest(BaseModel):
    conversationId: str
    ticker: str
    tradeDate: str
    prompt: str | None = None
    teamId: str | None = None
    selectedAnalysts: list[str] | None = None

File: mobile_h5/src/types/analysis.ts
export type MobileTaskStatus = 'idle' | 'pending' | 'running' | 'completed' | 'failed'

export interface AnalysisTaskSummary {
  taskId: string
  status: MobileTaskStatus
  symbol?: string
  currentStep?: string
  message?: string
  elapsedTime?: number
  remainingTime?: number
  reportId?: string
}

export interface CreateAnalysisTaskRequest {
  conversationId: string
  ticker: string
  tradeDate: string
  prompt?: string
  selectedAnalysts?: string[]
}
